### ISSUE TITLE
Unable to find image 'al-folio:latest' locally

### ISSUE THREAD

**Issue by kenluck2001** [iclr-blogposts/2024#44]
The download falls when trying to retrieve module from http://deb.debian.org/debian bookworm

Updating package manager, pruning docker image in my local and every step in [guide](https://stackoverflow.com/questions/62473932/at-least-one-invalid-signature-was-encountered) has failed me.

**To Reproduce**
1. Run ./bin/docker_run.sh
2. See error in additional context

**System (please complete the following information):**
 - OS: Ubuntu 16.04.6 LTS
 - Docker version: Docker version 18.09.6, build 481bc77

**Additional context**
```
Step 3/16 : RUN apt-get update -y
.
.
.
Err:3 http://deb.debian.org/debian-security bookworm-security InRelease
  At least one invalid signature was encountered.
Reading package lists...
W: GPG error: http://deb.debian.org/debian bookworm InRelease: At least one invalid signature was encountered.
E: The repository 'http://deb.debian.org/debian bookworm InRelease' is not signed.
W: GPG error: http://deb.debian.org/debian bookworm-updates InRelease: At least one invalid signature was encountered.
E: The repository 'http://deb.debian.org/debian bookworm-updates InRelease' is not signed.
W: GPG error: http://deb.debian.org/debian-security bookworm-security InRelease: At least one invalid signature was encountered.
E: The repository 'http://deb.debian.org/debian-security bookworm-security InRelease' is not signed.
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code
The command '/bin/sh -c apt-get update -y' returned a non-zero code: 100
Unable to find image 'al-folio:latest' locally
docker: Error response from daemon: pull access denied for al-folio, repository does not exist or may require 'docker login'.
See 'docker run --help'.
```

**Comment by fabianp** [comment]
Hi @kenluck2001 and thanks for the report. I just pushed a bunch of changes to the Dockerfile to make it more like the latest one from al-folio. Could you please pull the latest changes and let me know if the problem persists?

**Comment by kenluck2001** [comment]
@fabianp , thanks so much for getting back to me.

I am now failing for a different reason.

```
kenluck2001@kenluck2001-MS-7A71:~/Desktop/writing-paper/ICLR/2024-main$ ./bin/docker_run.sh
Sending build context to Docker daemon  15.75MB
Step 1/13 : FROM ubuntu:latest
latest: Pulling from library/ubuntu
aece8493d397: Pull complete 
Digest: sha256:2b7412e6465c3c7fc5bb21d3e6f1917c167358449fecac8176c6e496e5c1f05f
Status: Downloaded newer image for ubuntu:latest
 ---> e4c58958181a
Step 2/13 : ENV DEBIAN_FRONTEND noninteractive
 ---> Running in e0fbfb7949b1
Removing intermediate container e0fbfb7949b1
 ---> 65fbfb634f8a
Step 3/13 : RUN apt-get update -y && apt-get install -y --no-install-recommends     locales     imagemagick     ruby-full     build-essential     zlib1g-dev     jupyter-nbconvert     inotify-tools procps &&     apt-get clean && rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*
 ---> Running in db2106f378ba
Get:1 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]
Get:2 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Err:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
Get:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]
Err:1 http://archive.ubuntu.com/ubuntu jammy InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
Get:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [109 kB]
Err:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
Err:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
Reading package lists...
W: http://security.ubuntu.com/ubuntu/dists/jammy-security/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: http://security.ubuntu.com/ubuntu/dists/jammy-security/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: GPG error: http://security.ubuntu.com/ubuntu jammy-security InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
E: The repository 'http://security.ubuntu.com/ubuntu jammy-security InRelease' is not signed.
W: http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: GPG error: http://archive.ubuntu.com/ubuntu jammy InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
E: The repository 'http://archive.ubuntu.com/ubuntu jammy InRelease' is not signed.
W: http://archive.ubuntu.com/ubuntu/dists/jammy-updates/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: http://archive.ubuntu.com/ubuntu/dists/jammy-updates/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: GPG error: http://archive.ubuntu.com/ubuntu jammy-updates InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
E: The repository 'http://archive.ubuntu.com/ubuntu jammy-updates InRelease' is not signed.
W: http://archive.ubuntu.com/ubuntu/dists/jammy-backports/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: http://archive.ubuntu.com/ubuntu/dists/jammy-backports/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: GPG error: http://archive.ubuntu.com/ubuntu jammy-backports InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
E: The repository 'http://archive.ubuntu.com/ubuntu jammy-backports InRelease' is not signed.
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code
The command '/bin/sh -c apt-get update -y && apt-get install -y --no-install-recommends     locales     imagemagick     ruby-full     build-essential     zlib1g-dev     jupyter-nbconvert     inotify-tools procps &&     apt-get clean && rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*' returned a non-zero code: 100
kenluck2001@kenluck2001-MS-7A71:~/Desktop/writing-paper/ICLR/2024-main$ 

```

**Comment by fabianp** [comment]
Thanks for following up with us. From the suggestion here (https://askubuntu.com/questions/1429108/manually-add-key-the-following-signatures-couldnt-be-verified-because-the-publ), could you replace the line `FROM ubuntu:latest` in `Dockerfile` with `FROM ubuntu:18.04` and let us know if that solves the problem?

**Comment by kenluck2001** [comment]
@fabianp Thanks so much for the guide was helpful.Pulling from ubuntu 18.04 worked for the signing error. I am not so familiar with ruby packages.
However, can we have a script to ensure that ruby version, and rouge version is the recommended version in the Dockerfile

```
 ---> Running in ba65b4d5e079
ERROR:  Error installing jekyll:
	The last version of rouge (< 5.0, >= 3.0) to support your Ruby & RubyGems was 3.30.0. Try installing it with `gem install rouge -v 3.30.0` and then running the current command again
	rouge requires Ruby version >= 2.7. The current ruby version is 2.5.0.
ERROR:  Error installing bundler:
	The last version of bundler (>= 0) to support your Ruby & RubyGems was 2.3.26. Try installing it with `gem install bundler -v 2.3.26`
	bundler requires Ruby version >= 2.6.0. The current ruby version is 2.5.0.
```

**Comment by fabianp** [comment]
I also get that error but then somehow the build eventually succeeds:

```
Step 6/11 : RUN gem install jekyll bundler
 ---> Running in 677b189da2c4
ERROR:  Error installing jekyll:
        The last version of rouge (< 5.0, >= 3.0) to support your Ruby & RubyGems was 3.30.0. Try installing it with `gem install rouge -v 3.30.0` and then running the current command again
        rouge requires Ruby version >= 2.7. The current ruby version is 2.5.0.
ERROR:  Error installing bundler:
        The last version of bundler (>= 0) to support your Ruby & RubyGems was 2.3.26. Try installing it with `gem install bundler -v 2.3.26`
        bundler requires Ruby version >= 2.6.0. The current ruby version is 2.5.0.
Successfully installed webrick-1.8.1
Successfully installed unicode-display_width-2.5.0
Successfully installed terminal-table-3.0.2
Successfully installed safe_yaml-1.0.5
The command '/bin/sh -c gem install jekyll bundler' returned a non-zero code: 1
Configuration file: /srv/jekyll/_config.yml
            Source: /srv/jekyll
       Destination: /srv/jekyll/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
       Jekyll Feed: Generating feed for posts
       Imagemagick: Searching files in /srv/jekyll/assets/img/
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/10-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/10-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/10-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/11-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/11-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/11-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/12-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/12-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/12-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/7-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/7-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/7-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/8-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/8-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/8-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/9-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/9-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/9-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/iclr-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/iclr-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/2024-05-07-distill-example/iclr-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/ICLR-logo-dark-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/ICLR-logo-dark-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/ICLR-logo-dark-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/ICLR-logo-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/ICLR-logo-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/ICLR-logo-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/cg-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/cg-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/cg-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/cv-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/cv-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/cv-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/dd-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/dd-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/dd-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/fp-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/fp-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/fp-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/gg-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/gg-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/gg-1400.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/ls-480.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/ls-800.webp
       Imagemagick: Adding static file /srv/jekyll/_site/assets/img/organizers/ls-1400.webp
       Imagemagick: Generated 0 file(s)
         AutoPages: Disabled/Not configured in site.config.
        Pagination: Complete, processed 1 pagination page(s)
   Jekyll Diagrams: Command Not Found: mmdc
   Jekyll Diagrams: Command Not Found: mmdc
          Conflict: The following destination is shared by multiple files.
                    The written file may end up with unexpected contents.
                    /srv/jekyll/_site/news/announcement_1/index.html
                     - /srv/jekyll/news/announcement_1/index.html
                     - /srv/jekyll/_news/announcement_1.md
                    
          Conflict: The following destination is shared by multiple files.
                    The written file may end up with unexpected contents.
                    /srv/jekyll/_site/news/announcement_2/index.html
                     - /srv/jekyll/news/announcement_2/index.html
                     - /srv/jekyll/_news/announcement_2.md
                    
          Conflict: The following destination is shared by multiple files.
                    The written file may end up with unexpected contents.
                    /srv/jekyll/_site/news/announcement_3/index.html
                     - /srv/jekyll/news/announcement_3/index.html
                     - /srv/jekyll/_news/announcement_3.md
                    
                    done in 4.441 seconds.
 Auto-regeneration: enabled for '/srv/jekyll'
    Server address: http://0.0.0.0:8080/2024/
  Server running... press ctrl-c to stop.
```

if this does not work, have you tried legacy instructions [here](https://iclr-blogposts.github.io/2024/submitting/#method-2-using-jekyll-manually)? I just fixed some stuff in these instructions (8215275cf8d19dad6a10872ff0378c3d3f62d121), so if they failed previously perhaps you could give it another chance?

**Comment by kenluck2001** [comment]
Thanks so much. I got thing working using option 2. This is sufficient for me. Thanks @fabianp
